presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -130,6 +135,24 @@ protected InternalResourceGroup(Optional<InternalResourceGroup> parent, String n
     public ResourceGroupInfo getInfo()
     {
         synchronized (root) {
+            if (subGroups.isEmpty()) {
```

Comment:
I found this easier to read as there's an obvious termination condition and a recursive part. And the logic gets long so I figured separating them would be nice. But I'm ok with either way.